pkg/daemon/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,322 @@
+					},
+					svc.Status{State: svc.Running},
+				),
+				"test3-unmanaged": fake.NewFakeService(
+					"test3-unmanaged",
+					mgr.Config{
+						BinaryPathName: "test3-unmanaged",
+						Dependencies:   []string{},
+					},
+					svc.Status{State: svc.Running},
+				),
+			},
+			expectedRemovedServices:   []string{"test1", "test2", "test3-unmanaged"},
+			expectedRemainingServices: []string{},
+			preserveNode:              false,
+			configMapServices: []servicescm.Service{
+				{
+					Name:         "test1",
+					Command:      "test1 arg1",
+					Dependencies: []string{"test2", "test3-unmanaged"},
+					Bootstrap:    false,
+					Priority:     3,
+				},
+				{
+					Name:         "test2",
+					Command:      "test2 arg1 arg2",
+					Dependencies: []string{},
+					Bootstrap:    false,
+					Priority:     2,
+				},
+				{
+					Name:         "test3-unmanaged",
+					Command:      "test3-unmanaged",
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     0,
+				},
+			},
+		},
+	}
+	testAddrs := []net.Addr{&net.IPNet{IP: net.ParseIP("192.168.10.10")}}
+	for _, test := range testIO {
+		t.Run(test.name, func(t *testing.T) {
+			desiredVersion := "testversion"
+			// This ConfigMap's name must match with the given Node object's desired-version annotation
+			cm, err := servicescm.Generate(servicescm.NamePrefix+desiredVersion,
+				"openshift-windows-machine-config-operator", &servicescm.Data{test.configMapServices,
+					[]servicescm.FileInfo{}})
+			require.NoError(t, err)
+			clusterObjs := []client.Object{
+				// This is the node object that will be used in these test cases
+				&core.Node{
+					ObjectMeta: meta.ObjectMeta{
+						Name: "node",
+						Annotations: map[string]string{
+							nodeconfig.DesiredVersionAnnotation: desiredVersion,
+						},
+					},
+					Status: core.NodeStatus{
+						Addresses: []core.NodeAddress{
+							{Address: "192.168.10.10"},
+						},
+					},
+				},
+				cm,
+			}
+			fakeClient := clientfake.NewClientBuilder().WithObjects(clusterObjs...).Build()
+
+			winSvcMgr := fake.NewTestMgr(test.existingServices)
+			err = Execute(fakeClient, testAddrs, winSvcMgr, test.preserveNode)
+			require.NoError(t, err)
+			allServices, err := winSvcMgr.GetExistingServices()
+			require.NoError(t, err)
+
+			for _, svc := range test.expectedRemovedServices {
+				_, present := allServices[svc]
+				assert.Falsef(t, present, "service %s not removed as expected", svc)
+			}
+			for _, svc := range test.expectedRemainingServices {
+				_, present := allServices[svc]
+				assert.Truef(t, present, "service %s not present as expected", svc)
+			}
+			// Depending on if the test is an upgrade cleanup case, check if the node object has been deleted or not
+			nodes := &core.NodeList{}
+			require.NoError(t, fakeClient.List(context.TODO(), nodes))
+			if test.preserveNode {
+				assert.NotZero(t, len(nodes.Items))
+			} else {
+				assert.Zero(t, len(nodes.Items))
```

Comment:
I suggest using `assert.Empty()` and `NotEmpty()`. The `len()` calls are unecessary